invenio_app_ils/ui/src/invenio_app_ils/common/api/documentRequests/documentRequest.js

```diff
@@ -12,26 +12,20 @@ const create = async data => {
   return response;
 };
 
-const get = docRequestPid => {
-  return http.get(`${documentRequestURL}${docRequestPid}`).then(response => {
-    response.data = serializer.fromJSON(response.data);
-    return response;
-  });
+const get = async docRequestPid => {
+  const response = await http.get(`${documentRequestURL}${docRequestPid}`);
+  response.data = serializer.fromJSON(response.data);
+  return response;
 };
 
 const del = async docRequestPid => {
   const response = await http.delete(`${documentRequestURL}${docRequestPid}`);
   return response;
 };
 
-const patch = async (docRequestPid, ops) => {
-  const response = await http.patch(
-    `${documentRequestURL}${docRequestPid}`,
-    ops,
-    {
-      headers: { 'Content-Type': 'application/json-patch+json' },
-    }
-  );
+const performAction = async (docRequestPid, action, data) => {
```

Comment:
for reviewers, do you prefer here to exposed a `performAction` passing the param `action` `accept/reject`
-- or --
`performAccept()` and `performReject`
?

In loans, we have `performAction` because we don't know the action name, we get it from the `links` of the loan, so it must be generic...

Reply:
Personally I am in favour of being explicit with these names :) It makes me understand the function without checking the code!